### PR TITLE
test(pipelines): Add openshift nightly job

### PR DESCRIPTION
**Pull request by nshankar13**
Signed-off-by: nshankar13 <nshankar@microsoft.com>

**Description**: Add nightly job for e2es on OpenShift.

**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| New Functionality          | [ ] |
| Documentation              | [ ] |
| Install                    | [ ] |
| CLI Tool                   | [ ] |
| Certificate Management     | [ ] |
| Control Plane              | [ ] |
| Ingress                    | [ ] |
| Egress                     | [ ] |
| Networking                 | [ ] |
| Observability              | [ ] |
| SMI Policy                 | [ ] |
| Sidecar Injection          | [ ] |
| Security                   | [ ] |
| Upgrade                    | [ ] |
| Tests                      | [X] |
| CI System                  | [X] |
| Demo                       | [ ] |
| Performance                | [ ] |
| Other                      | [ ] |

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project? No
    -   Did you notify the maintainers and provide attribution?

1. Is this a breaking change? No

### CODE REVIEW

File: .github/workflows/openshift-nightly.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  test:
+    name: OpenShift Nightly Job
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout v2
+        uses: actions/checkout@v2
+      - name: Authenticate and set context
+        uses: redhat-actions/oc-login@v1
+        with:
+          openshift_server_url: ${{ secrets.OPENSHIFT_SERVER }}
```
Comment (shalier) [inline .github/workflows/openshift-nightly.yml:16]: do the secrets have to be updated ever?

Comment (nshankar13) [reply .github/workflows/openshift-nightly.yml:16]: The token we use is a permanent token, so it should not need to be updated. @ksubrmnn

**Review by SanyaKochhar — APPROVED** [review]

File: .github/workflows/openshift-nightly.yml (hunk 1 above)
Comment (ksubrmnn) [reply .github/workflows/openshift-nightly.yml:16]: Yep that's correct! The server and token should stay the same

**Review by ksubrmnn — APPROVED** [review]
Amazing! Thanks Niranjan!

**Comment by nshankar13** [comment]
@mergifyio backport release-v0.9

**Pull request merged**